### PR TITLE
example: Update `example_cli` and retire old nursery crates

### LINKED ISSUE #1469: Remove nursery/tmp_plan
The planning module has been added to `rust-miniscript` so this crate is no longer needed.

https://docs.rs/miniscript/latest/miniscript/plan/index.html

### PR DESCRIPTION
**Pull request by ValuedMammal**
- Adds two commands `init` and `generate`. Loading database doesn't require descriptors
- Replaces `send` command with `psbt` (new, sign, and extract). Supports tap key spend
- Uses `bdk-coin-select` and miniscript `plan` module

fixes #1469 
partially addresses #1462

### Notes to the reviewers

Note the `example_cli` lib now defines the `ChangeSet` and `Anchor` type whereas before these were generic

### Changelog notice

* Improvements to `example_cli` that include generating descriptors and creating PSBTs

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by storopoli** [comment]
We could merge this with #1442 and then commit it as a single PR?

If favorable I could rebase #1442 and then you can commit on top of that? Or I can do the opposite as well...

**Comment by ValuedMammal** [comment]
1442 can be merged anytime and from what I can tell doesn't conflict with this one

**Comment by rustaceanrob** [comment]
Nice. This cleans things up quite a bit.

**Comment by ValuedMammal** [comment]
* Rebased onto 17a9850cba06db837b82cef1dbaf2ef1e342f7ec
* For now I'm using `bdk_wallet::ChangeSet` for the example-cli
* Added back support for single descriptor use
* CI failed to build msrv

File: example-crates/example_cli/Cargo.toml [hunk 1]
```diff
@@ -7,11 +7,11 @@ edition = "2021"
 
 [dependencies]
 bdk_chain = { path = "../../crates/chain", features = ["serde", "miniscript"]}
+bdk_wallet = { path = "../../crates/wallet" }
```
Comment (LLFourn) [inline example-crates/example_cli/Cargo.toml]: Can we not depend on `bdk_wallet` for its changeset. Just redefine it here for now. I know it's friction and seems like a waste of time but the point of the `example_cli` examples is more to demonstrate to us as developers what you have to do to make a wallet with `bdk_chain` + the backends. There is of course still way too much you have to do but it's better not to hide it by depending on `bdk_wallet`.

**Review by LLFourn — COMMENTED** [review]
ConceptACK this looks great.

**Review by evanlinjin — APPROVED** [review]
ACK 771f6b9c97a70be2e8d2da5216f3b686295861bd

A lot better than what we had before. Made the examples a lot more useful.

**Comment by ValuedMammal** [comment]
Thanks @evanlinjin @LLFourn 

We're currently limited to creating PSBTs for `wpkh` and `tr` descriptors because I got stuck on trying to add assets/keys for `wsh` (can still sync and derive addresses though).

**Comment by evanlinjin** [comment]
@ValuedMammal I think that's fine for now, can add them later

**Pull request merged**